draft-nygren-altsvc-fixes.md

```diff
@@ -123,22 +123,79 @@ Two options (and there may be more) include:
 
 ## Better Interactions with HTTPS Record
 
-This was deferred out of the HTTPS RR draft.  There are a number of design
-options here, but requirements and pros/cons will want to be discussed in-detail
-before proposing designs.  Supporting ECH and Alt-Svc together is a primary
-goal.
-
-An important item here is which takes precedence, providing safe and
-time-bounded ways to allow Alt-Svc to take precedence over HTTPS records.
-Alt-Svc has the ability to be delivered in a user-specific manner -- useful
-operationally, but potentially problematic for privacy.  HTTPS records can be
-easily revalidated, which is more difficult with an Alt-Svc record.
+Interaction between Alt-Svc and the HTTPS RR was a difficult problem in the
+SVCB/HTTPS draft, as dnsop is not the ideal place to update Alt-Svc.
 
 Providing a way for Alt-Svc to act as AliasMode references to HTTPS SvcMode
 records seems like one clean way for interaction in that it avoids needing to
 duplicate SVCB in Alt-Svc.  We would still need to address time-bounding and
 trust considerations.
 
+### Relative Priority
+
+Alt-Svc and HTTPS records have different interaction and trust models.  Alt-Svc
+is provided in-connection, and might be targeted to the specific client instead
+of simply based on the client IP address. It is also received over an
+authenticated channel, so that the Alt-Svc entry is known to come from the
+origin server.
+
+An HTTPS record is published in the DNS, and can be targeted only to the extend
+that any DNS record might be targeted based on the client subnet.  In the
+absence of DNSSEC signing and verification, information from the DNS must be
+considered untrusted.
+
+Alt-Svc records are at best cached from the last interaction with the server,
+while HTTPS records can be retrieved when a client is preparing to make a
+connection to the server.
+
+When the client is preparing to make a connection, the Alt-Svc records are more
+trusted and more tailored, while the HTTPS records are more current.  Should a
+client not do HTTPS queries when it has a usable Alt-Svc record?  Should the
+HTTPS records override Alt-Svc?  Or should the two mechanisms interact in some
+way?
+
+{{Section 8.3 of SVCB}} resolves the question this way:
+
+> Clients that implement support for both Alt-Svc and HTTPS records SHOULD
+> retrieve any HTTPS records for the Alt-Svc alt-authority, and ensure that
+> their connection attempts are consistent with both the Alt-Svc parameters and
+> any received HTTPS SvcParams. If present, the HTTPS record's TargetName and
+> port override the alt-authority.
+
+That is, if Alt-Svc provides an alternative host, it changes the hostname being
+resolved away from the origin stated in the URL.  Once those HTTPS records are
+retrieved, the capabilities advertised in Alt-Svc filter the endpoints
+advertised in the DNS.  However, only the endpoints advertised in DNS are
+actually used.
```

Comment:
```suggestion
advertised in the DNS.  However, only the endpoints advertised in DNS are
actually used.  (For example, Encrypted Client Hello
configuration only lives in HTTPS records and not
also as an Alt-Svc parameter.)
```

Reply:
This parenthetical doesn't seem related to the paragraph.  I'll go back and add the point elsewhere.